### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by lRebornsl**
# 🎏   In this *Milestone*  🙋‍♂️⁉:
- ✅ Follow SOLID principles.
- ✅ No linters error.
- ✅ Follow the best GitHub flow.

### CODE REVIEW

File: list_hud.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      @books.add_book
+    when 5
+      @rental_handler.rent_book
+    when 6
+      @rental_handler.list_rentals
+    else
+      puts 'Invalid choice'
+    end
+  end
+end
```
Comment (danmainah) [inline list_hud.rb]: - [ ][optional] @Team great work adding a class to deal with UI but it makes the app.rb redundant as its only work is to import this class then pass it to main,rb file, I would highly advise moving this functionality to App.rb file.

File: persons_handler.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Specialization: '
+    specialization = gets.chomp
+    @persons << Teacher.new(name, age, specialization)
+    puts 'Teacher created successfully'
+  end
+end
```
Comment (danmainah) [inline persons_handler.rb]: - [x] @Team Great work adding methods to add new persons, however, to achieve the `Single-responsibility Principle` all requests for input and input validation should've been placed in the main.rb file and/or classes/methods that are responsible for requesting or validating input. You can add a new file whose main role is to ensure the user can add new persons.

File: books_handler.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def add_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
```
Comment (danmainah) [inline books_handler.rb]: - [x] @Team Great work adding a method to add new book, however, to achieve the Single-responsibility Principle all requests for input and input validation should've been placed in the main.rb file and/or classes/methods that are responsible for requesting or validating input. You can add a new file whose main role is to ensure the user can add new persons.

File: rental_handler.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    puts 'Select a person from the following list by number'
+    @persons.each_with_index do |person, index|
+      puts "#{index}. Name: #{person.name} Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+    puts 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_index], @persons[person_index])
+    puts 'Book rented successfully'
+  end
```
Comment (danmainah) [inline rental_handler.rb]: - [x] @Team Great work adding a method to add a new rental, however, to achieve the Single-responsibility Principle all requests for input and input validation should've been placed in the main.rb file and/or classes/methods that are responsible for requesting or validating input. You can add a new file whose main role is to ensure the user can add new persons.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @lRebornsl and @Abdelaziz-Mahdi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Refactor code to ensure that main.rb is only to start your code
✔️  Ensured the app was working correctly after refactoring.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: rental_handler.rb (hunk 4 above)
Comment (lRebornsl) [reply rental_handler.rb]: Solved

File: books_handler.rb (hunk 3 above)
Comment (lRebornsl) [reply books_handler.rb]: Solved

File: persons_handler.rb (hunk 2 above)
Comment (lRebornsl) [reply persons_handler.rb]: Solved

File: list_hud.rb (hunk 1 above)
Comment (lRebornsl) [reply list_hud.rb]: Solved

**Review by frank1738 — APPROVED** [review]
Hi Team 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Good work addressing the single responsibility issue
✅ App works as expected

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**